src/sympc/tensor/static.py

```diff
@@ -11,18 +15,18 @@ def argmax(self, dim=None, keepdim=False, one_hot=False) -> "MPCTensor":
         keepdim: when one_hot is true, keep all the dimensions of the tensor
         one_hot: return the argmax as a one hot vector
     """
-    # implementation pseudo code 
+    # implementation pseudo code
 
     x = self.flatten() if dim is None and len(self.shape) > 1 else self
-    # for each share in MPCTensor 
+    # for each share in MPCTensor
     #   do the algorithm protrayed in paper (helper_argmax_pairwise)
-    #   results in creating two matrices and substracting them 
-    args = [dim]
-    shares = parallel_execution(helper_argmax_pairwise, self.session.parties)(args)    
-    # then create an MPCTensor tensor based on this results per share 
+    #   results in creating two matrices and substracting them
+    args = [[share_ptr_tensor, dim] for share_ptr_tensor in self.share_ptrs]
```

Comment:
Did removing the cyclic import and adding this line already solve the issue for you @gmuraru concerning the remote call? (That's also the version I tried it locally and it didn't work -- is it a problem on my end maybe?)